libcodechecker/server/client_db_access_handler.py

```diff
@@ -640,51 +648,72 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
                     # There is no difference.
                     return results
 
-            q = session.query(Report,
-                              File,
-                              ReviewStatus) \
-                .filter(Report.run_id.in_(run_ids)) \
-                .outerjoin(File, Report.file_id == File.id) \
-                .outerjoin(ReviewStatus,
-                           ReviewStatus.bug_hash == Report.bug_id) \
-                .filter(filter_expression)
+            isUnique = report_filter is not None and \
+                       report_filter.isUnique
+
+            if isUnique:
+                q = session.query(Report.bug_id,
+                                  Report.checker_id,
+                                  Report.checker_message,
+                                  Report.severity,
+                                  File.filename,
+                                  ReviewStatus) \
+                           .distinct(Report.bug_id,
```

Comment:
I would leave a TODO here. Is this deterministic which bug will be chosen from the ones with the same bug id? Maybe in the future, we would like to get the one with the shortest path or have some other criterion.